Cache creation & integration.

filipay opened the pull request:
Added some inital tests.
Builds now in Travis CI.
Refactoring and focus on Dependancy Injection for testing.

filipay committed:
Merge pull request #1 from filipay/testing

Testing

filipay committed:
Deleted *.db, due to sensitive info. Added models folder

filipay committed:
Network scan always updates the DB now.

filipay committed:
Seperated behaviours between watching and monitoring.

filipay committed:
DB no longer handles caching. Metrics do.
Added /live and /all to /scan

filipay committed:
Added Winston logger.
Moved classes around.

filipay committed:
Added PUT for editing devices.

filipay committed:
Changed behaviour of watch.js.
Now you can set different type of watches for diff intervals.
FIX: PUT came through but didn't update value

HughMcDonnell committed:
added networkspeeds via speedtest-net

filipay committed:
Work started on dashboard. Live results display by default.
Small tweaks at metrics.
Disabled watch in /routes/api.js

filipay committed:
Added 'Loading..', to notify users that we're waiting on data.

filipay committed:
Changed from /scan to /devices.
Updated the behaviour of PUT /devices. Now updates cache! (WOW)

filipay committed:
Changed to dynamic link

filipay committed:
Deleted unnecessary html.

filipay committed:
Created a HybridCache. Tries memory, then DB.
Added a simple test. Needs to be expanded.

filipay committed:
Created a HybridCache. Tries memory, then DB.
Added a simple test. Needs to be expanded.

filipay committed:
Merge branch 'dev-filip' of https://github.com/filipay/monitroom into dev-filip

filipay committed:
Forgot to add file.. I hope everything is okay.. I was playing with git

filipay committed:
Metrics now use cache. Increased response speed.
Devices all and live are displaying saved data.

filipay committed:
Metrics.networkScan now needs a paramter for live or all.
If PUT request comes back with an err, give 403 Forbidden.
Cache.create() (previously .init()) now uses fetchAll implemented by a given DB.
Cache.getAll() now merges latest data with old.

filipay committed:
Added support for Travis CI
Started working on testing

filipay committed:
Added more tests. Metrics now allows for Dependancy Injection

filipay committed:
Testing React.
Started work on proper logging.
Query logs kinda works.
Renamed tests.

HughMcDonnell commented on package.json:
```diff
@@ -1,10 +1,12 @@
+    "start": "node ./bin/www",
+    "test": "./node_modules/mocha/bin/mocha"
   },
+  "license": "MIT",
```
Is this norm?

filipay replied:
As far as I know yes

HughMcDonnell commented on public/javascripts/dashboard.js:
```diff
@@ -1,30 +1,42 @@
+  var td = $('.td-template').empty().clone();
+  var tr = $('.tr-template').empty().clone();
+  var index = 0;
+  $('.tbody-template').html('<h3>Loading...<\h3>');
```
Should you setup a CONSTANTS class for strings like this?

filipay replied:
This is more or less deprecated with my latest commit

HughMcDonnell commented on public/javascripts/table.jsx:
```diff
@@ -0,0 +1,82 @@
+var DeviceRow = React.createClass({
+  lastSeen : function(timestamp) {
+    var second = 1000;
```
These could also be moved to proposed CONSTANTS class.

filipay replied:
I dunno if these should be seen as constants... They're more like singleton's I think. Also a file called constants would get pretty messy and have all sorts of constants. IMO having these in a table class is fine but more input is welcome

HughMcDonnell commented on public/javascripts/table.jsx:
```diff
@@ -0,0 +1,82 @@
+      index++;
+    });
+    return (
+      <table className="table table-striped">
```
Could this stuff be moved some where else to help make it easier to maintain? (separate code from strings)

filipay replied:
I'm not sure what you mean.. Why should this code be moved?

HughMcDonnell reviewed (approved):
Looks good! Lots of welcome changes. 

General (small) comment in relation to style and formatting: maybe setup some style checking/linting because your style is inconsistent slowing comprehension.

e.g. conditional logic across various files:
if ( cond) 
if (cond) 
if ( cond )

filipay committed:
Added some simple Logging.
Querying logs can be used through the GET API.
Some refactoring and helper methods added.
